include/pybind11/stl_bind.h

```diff
@@ -326,6 +327,36 @@ template <typename Vector, typename Class_> auto vector_if_insertion_operator(Cl
     );
 }
 
+// Provide the buffer interface for vectors if we have data() and we have a format for it
+// GCC seems to have "void std::vector<bool>::data()" - doing SFINAE on the existence of data() is insufficient, we need to check it doesn't return void
```

Comment:
If the goal is just to detect the ``bool`` special case, wouldn't it be more transparent to SFINAE on that?